nextcord/embeds.py

```diff
@@ -427,17 +456,76 @@ def set_image(self: E, url: MaybeEmpty[Any]) -> E:
         -----------
         url: :class:`str`
             The source URL for the image. Only HTTP(S) is supported.
+        file: Optional[:class:`File`]
+            A file to use for the image.
+
+            .. versionadded:: 2.1
         """
+        from . import InvalidArgument
 
         if url is EmptyEmbed:
             try:
                 del self._image
             except AttributeError:
                 pass
+
+        if file is EmptyEmbed:
+            self._local_files.pop("image", None)
+
+        elif url is not EmptyEmbed and file is not EmptyEmbed:
```

Comment:
This is already checked by elif
```suggestion
        elif url is not EmptyEmbed:
```

Reply:
I'll change it to an `if`, as they are actually meant to be 3 separate logical blocks.
1) Reset existing url image
2) Reset existing file image
3) Set a new image. You can only set 1 image, thus the check